controllers/ccruntime_controller.go

```diff
@@ -651,6 +590,42 @@ func (r *CcRuntimeReconciler) processDaemonset(operation DaemonOperation) *appsv
 		containerCommand = r.ccRuntime.Spec.Config.UninstallCmd
 	}
 
+	var debug = strconv.FormatBool(r.ccRuntime.Spec.Config.Debug)
+
+	var defaultShim = ""
+	var createDefaultRuntimeClass = "false"
+	if r.ccRuntime.Spec.Config.DefaultRuntimeClassName != "" {
```

Comment:
Since we expect the DefaultRuntimeClassName to only contain a single value with `kata-` prefix, may be it's better to have a strict check. Something like this
```
if strings.HasPrefix(r.ccRuntime.Spec.Config.DefaultRuntimeClassName, "kata-") {
    // Remove the "kata-" prefix from DefaultRuntimeClassName
    defaultShim = strings.TrimPrefix(r.ccRuntime.Spec.Config.DefaultRuntimeClassName, "kata-")
    createDefaultRuntimeClass = "true"
}
```